### PR TITLE
feat/game-history

### PR DESCRIPTION
**Pull request by erick-medeiros**
* game history
* fixing ball collision with paddle
* downgrade typescript for best compatibility with eslint
* change location of game invites
* leaderboard use users info

eandre-f

leaderboard

### CODE REVIEW

File: backend/src/core/entities/user.entity.ts [hunk 1]
```diff
@@ -127,8 +133,11 @@ export class UserEntity {
   @OneToMany(() => BlockListEntity, (block) => block.blockedBy)
   blockedUsers: BlockListEntity[];
 
-  @ManyToMany(() => GameEntity)
-  games: GameEntity[];
+  @OneToMany(() => GameEntity, (game) => game.playerOne)
+  gamePlayerOne: GameEntity[];
+
+  @OneToMany(() => GameEntity, (game) => game.playerTwo)
+  gamePlayerTwo: GameEntity[];
```
Comment (LucasKuhn) [inline backend/src/core/entities/user.entity.ts:140]: Não estamos usando isso
```suggestion
```

File: backend/src/game/game.gateway.ts [hunk 2]
```diff
@@ -178,4 +178,9 @@ export class GameGateway
   ) {
     this.gameService.movePlayers(userId, body);
   }
+
+  @SubscribeMessage('getGameHistory')
+  async handleGetGameHistory(@MessageBody() userId: number) {
+    return await this.gameService.getGameHistory(userId);
```
Comment (LucasKuhn) [inline backend/src/game/game.gateway.ts:184]:
```suggestion
    return await this.gameService.getAllGames(userId);
```

File: backend/src/game/game.service.ts [hunk 3]
```diff
@@ -200,33 +253,56 @@ export class GameService {
...
     this.server
       ?.to(`game:${gameId}`)
-      .emit('updatePlayerPosition', this.games[gameId].players);
+      .emit('updatePlayerPosition', [
+        this.gamesInMemory[gameId].playerOne,
+        this.gamesInMemory[gameId].playerTwo,
+      ]);
+  }
+
+  async getGameHistory(userId: number) {
```
Comment (LucasKuhn) [inline backend/src/game/game.service.ts:301]:
```suggestion
  async getAllGames(userId: number) {
```

File: backend/src/game/game.service.ts [hunk 4]
```diff
@@ -200,33 +253,56 @@ export class GameService {
...
+    const history = await this.gameRepository.find({
+      relations: { playerOne: true, playerTwo: true },
+      where: [{ playerOne: { id: userId } }, { playerTwo: { id: userId } }],
+    });
+    return history;
```
Comment (LucasKuhn) [inline backend/src/game/game.service.ts:306]:
```suggestion
  async getAllGames(userId: number) {
    const games = await this.gameRepository.find({
      relations: { playerOne: true, playerTwo: true },
      where: [{ playerOne: { id: userId } }, { playerTwo: { id: userId } }],
    });
    return games;
```

**Review by ichmi — APPROVED** [review]

**Review by LucasKuhn — APPROVED** [review]
LGTM!

**Review by LucasKuhn — APPROVED** [review]

**Pull request merged**